pkg/operator/csr/csr_approver.go

```diff
@@ -175,19 +176,30 @@ func getCertApprovalCondition(status *certapiv1.CertificateSigningRequestStatus)
 }
 
 type ServiceAccountApprover struct {
-	saGroups        sets.Set[string] // saGroups is the set of groups for the SA expected to have created the CSR
-	saName          string
-	expectedSubject string
+	saGroups         sets.Set[string] // saGroups is the set of groups for the SA expected to have created the CSR
+	saName           string
+	expectedSubjects []string
```

Comment:
Instead of a string slice, should this be a `sets.Set[string]` to:
- Prevent duplicates (should duplicates be allowed?)
- Have faster lookup than `slices.Contains`
- Not require a new import (even though it is a standard library import)

Reply:
For the intended use case, a slice would be sufficient. but sure let's got with a set to keep consistency.